packages/clerk-js/src/ui/utils/createCustomMenuItems.tsx

```diff
@@ -0,0 +1,241 @@
+      }
+      if (b.id === USER_BUTTON_ITEM_ID.MANAGE_ACCOUNT) {
+        return 1;
+      }
+      return 0;
+    });
+  }
+
+  return { menuItems: allItems };
+};
+
+const getUserButtonDefaultMenuItems = () => {
+  const INITIAL_MENU_ITEMS = [
+    {
+      name: localizationKeys('userButton.action__manageAccount'),
+      id: USER_BUTTON_ITEM_ID.MANAGE_ACCOUNT as 'manageAccount',
+      icon: CogFilled as React.ComponentType,
+    },
+    {
+      name: localizationKeys('userButton.action__signOut'),
+      id: USER_BUTTON_ITEM_ID.SIGN_OUT as 'signOut',
+      icon: SignOut as React.ComponentType,
+    },
+  ];
+
+  const validReorderItemLabels: string[] = INITIAL_MENU_ITEMS.map(r => r.id);
+
+  return { INITIAL_MENU_ITEMS, validReorderItemLabels };
+};
+
+const isValidCustomMenuItem = (item: CustomMenuItem, validReorderItemLabels: string[]): item is CustomMenuItem => {
+  return (
+    (isCustomMenuLink(item) ||
+      isCustomMenuItem(item) ||
+      isCustomMenuItemWithOpen(item) ||
+      isReorderItem(item, validReorderItemLabels)) &&
+    typeof item.label === 'string'
+  );
+};
+
+const isCustomMenuItem = (ci: CustomMenuItem): ci is CustomMenuAction => {
+  return !!ci.label && !!ci.onClick && !ci.mount && !ci.unmount && !!ci.mountIcon && !!ci.unmountIcon;
+};
+
+const isCustomMenuItemWithOpen = (ci: CustomMenuItem): ci is CustomMenuActionWithOpen => {
+  return !!ci.label && !!ci.open && !ci.mount && !ci.unmount && !!ci.mountIcon && !!ci.unmountIcon;
+};
+
+const isCustomMenuLink = (ci: CustomMenuItem): ci is CustomMenuLink => {
+  return !!ci.href && !!ci.label && !ci.mount && !ci.unmount && !!ci.mountIcon && !!ci.unmountIcon;
+};
+
+const isReorderItem = (ci: CustomMenuItem, validItems: string[]): ci is MenuReorderItem => {
+  return !ci.mount && !ci.unmount && !ci.mountIcon && !ci.unmountIcon && validItems.some(v => v === ci.label);
+};
+
+const checkForDuplicateUsageOfReorderingItems = (customMenuItems: CustomMenuItem[], validReorderItems: string[]) => {
+  const reorderItems = customMenuItems.filter(ci => isReorderItem(ci, validReorderItems));
+  reorderItems.reduce((acc, ci) => {
+    if (acc.includes(ci.label)) {
+      console.error(
+        `Clerk: The "${ci.label}" item is used more than once when reordering pages. This may cause unexpected behavior.`,
+      );
+    }
+    return [...acc, ci.label];
+  }, [] as string[]);
+};
+
+const warnForDuplicateLabels = (items: CustomMenuItem[]) => {
+  const labels = items.map(item => item.label);
+  const duplicates = labels.filter((label, index) => labels.indexOf(label) !== index);
+  if (duplicates.length > 0) {
+    console.warn(`Clerk: Duplicate custom menu item labels found: ${duplicates.join(', ')}`);
+  }
+};
+
+const sanitizeCustomLinkURL = (url: string): string => {
+  if (!url) {
+    throw new Error('Clerk: URL is required for custom links');
+  }
+  if (isValidUrl(url)) {
+    return url;
+  }
+  return (url as string).charAt(0) === '/' ? url : `/${url}`;
+};
```

Comment:
This seems like a duplicate from `createCustomPages`, we can move it into a shared file

Reply:
Done